codec-http/src/main/java/io/netty/handler/codec/http/QueryStringDecoder.java

```diff
@@ -338,18 +549,29 @@ private static String decodeComponent(String s, int from, int toExcluded, Charse
         if (len <= 0) {
             return EMPTY_STRING;
         }
-        int firstEscaped = -1;
-        for (int i = from; i < toExcluded; i++) {
-            char c = s.charAt(i);
-            if (c == '%' || c == '+' && !isPath) {
-                firstEscaped = i;
-                break;
-            }
-        }
+        int firstEscaped = getFirstEscaped(s, from, toExcluded, isPath);
         if (firstEscaped == -1) {
             return s.substring(from, toExcluded);
         }
 
+        return decodeEscapedComponent(s, from, toExcluded, charset, isPath, firstEscaped, len);
+    }
+
+    private static int getFirstEscaped(String s, int from, int toExcluded, boolean isPath) {
+        int cutOff = !isPath? '+' : '%';
```

Comment:
Let's make this easier to read 
```suggestion
        int cutOff = isPath? '%' : '+' ;
```